code/modules/events/spacevine.dm

```diff
@@ -16,7 +16,13 @@
 	for(var/area/hallway/A in world)
 		for(var/turf/F in A)
 			if(F.Enter(SV))
-				turfs += F
+				var/valid = TRUE
+				for(var/mob/living/L in view(F, 7))
```

Comment:
This is going to be extremely awful performance

Reply:
luckily this whole routine will only run once, if at all, per round

Reply:
It's still potentially going to cause a noticeable freeze

Reply:
Something with less performance impact may be checking the `view` when it decides on a turf, and rerolling the random turf selection if that isn't a "valid" placement. While you technically could make it do this infinitely, it might be better to give it a maximum number of retries. While this won't _completely_ fix the problem, it will make situations where vines spawn near people _much_ less common.

Reply:
is 50 retries ok?

Reply:
A little much, but better than checking every tile by far. My personal recommendation would be around 10 to 20, but you can consult with an actual maintainer on their opinion of an option.